airflow/models/dag.py

```diff
@@ -227,7 +230,8 @@ def __init__(
         params: Optional[Dict] = None,
         access_control: Optional[Dict] = None,
         is_paused_upon_creation: Optional[bool] = None,
-        jinja_environment_kwargs: Optional[Dict] = None
+        jinja_environment_kwargs: Optional[Dict] = None,
+        tags: List[str] = []
```

Comment:
Mutable default arguments are a no-no, please change to `None` and instantiate in the `__init__` (see e.g. https://blog.florimond.dev/python-mutable-defaults-are-the-source-of-all-evil)